module/VuFindTheme/src/VuFindTheme/Initializer.php

```diff
@@ -303,16 +306,29 @@ protected function getThemeOptions($currentTheme)
         $options = [];
         if (isset($this->config->selectable_themes)) {
             $parts = explode(',', $this->config->selectable_themes);
+            $foundSelected = false;
             foreach ($parts as $part) {
                 $subparts = explode(':', $part);
                 $name = trim($subparts[0]);
                 $desc = isset($subparts[1]) ? trim($subparts[1]) : '';
                 $desc = empty($desc) ? $name : $desc;
+                // Easiest and most accurate way to pick a selected theme is to check
+                // if the name matches the current value of the ui cookie:
+                $selected = $this->cookieManager->get('ui') === $name;
```

Comment:
The issue which created the need for this whole PR is that the `ui` cookie is set to an *alias* (e.g. `standard`, `mobile` or one of the custom aliases set up in config.ini) while `$currentTheme` is set to the actual name of the theme we get when resolving the alias. Since the drop-down menu lists the aliases and not the actual themes, we need to use the raw version in order to highlight things correctly.

Reply:
Argh. I think it would be better for pickTheme to return the actual theme and the alias, but this is fine for now to keep changes to minimum.

Reply:
But I still want the cookie getter outside of the loop. I know it doesn't count for much, but still.

Reply:
(It is a good idea to refactor the get call outside of the loop, though -- I've done that).

Reply:
Also note that cookieManager gets updated with the modified value -- it's not reading from the cookie after values are changed.

Reply:
But yes, this is an ugly mess that should be refactored more neatly in a major revision... just trying to find something that works for the patch first!